plugin/pkg/shoot/validator/admission.go

```diff
@@ -569,6 +575,55 @@ func (c *validationContext) validateProvider() field.ErrorList {
 	return allErrs
 }
 
+func (c *validationContext) validateAPIVersionForRawExtensions() field.ErrorList {
+	var allErrs field.ErrorList
+
+	if usesInternalVersion, gvk := checkRawExtension(c.shoot.Spec.Provider.InfrastructureConfig); usesInternalVersion {
```

Comment:
How about naming the function `usesInternalVersion` instead of the variable that holds the result? The variable then could just be named `ok` (as usual for such short-lived boolean values in Go)
```suggestion
	if ok, gvk := usesInternalVersion(c.shoot.Spec.Provider.InfrastructureConfig); ok {
```

Reply:
done.